### ISSUE TITLE
torch.clamp doesn't propagate nan values from boundaries

### ISSUE THREAD

**Issue by ngimel** [pytorch/pytorch#76795]
`torch.clamp_min` and `torch.maximum` are supposed to produce the same results, yet when boundary tensor contains `nan`, they are propagated for `maximum` and not propagated for `clamp/clamp_min`:
```
In [7]: a=torch.zeros(32)

In [8]: maxes=torch.ones(32)

In [9]: a[16:]=float('nan')

In [10]: maxes[:15]=float('nan')

In [11]: torch.maximum(a,maxes)
Out[11]: 
tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, 1., nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])

In [12]: torch.clamp_min(a,maxes)
Out[12]: 
tensor([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan])
```
This happens both on cuda and cpu, both vectorized and non-vectorized kernels. 
For primTorch decompositions, one of these functions will be expressed through the other, so behavior should match. Jax and numpy propagate nans from boundary tensors for `(j)np.clip`.

cc @ezyang @mruberry @ngimel

**Comment by vadimkantorov** [comment]
does clip_min propagate gradients to the `maxes` (if tensor argument maxes is used and it requires_grad)? if yes, probably clamp_min and maximum should become aliases

**Comment by ngimel** [comment]
@vadimkantorov see #76522, there's a thoughtful discussion going on about this. For the record, jax's gradients for relu (that's just implemented via maximum) and maximum itself are different, and our maximum gradients are not great either, because they don't differentiate between scalar/tensor args and requires gradness, and always spread the grads.

**Comment by Lezcano** [comment]
Fixed in https://github.com/pytorch/pytorch/pull/77306
